scripts/make-official-release.sh

```diff
@@ -0,0 +1,354 @@
+#!/bin/bash
+
+# Make official release
+# =====================
+#
+# This script helps RxPlayer developers to make a new official RxPlayer release.
+#
+# To use it:
+#
+#   1. Be sure that you're either on the `stable` branch (if this is a "patch"
+#      release in terms of SEMVER) or on the `dev` branch (if this is a "minor"
+#      SEMVER release)
+#
+#   2. Call this script, you may optionally provide the wanted version as an
+#      argument to this script. If no argument is provided, the script will ask
+#      for your wanted version number.
+#
+#   3. Wait for all the commands to finish and stay attentive to the output. It
+#      will ask you for inputs at various steps and tell you what you need to
+#      know to make a release.
+
+set -e
+
+# Log a line prefixed with our script's name
+log() {
+  echo "---- RxPlayer Release Script ----   $1"
+}
+
+# Log a line prefixed with our script's name after an empty line then sleep a little
+emphasized_log() {
+  echo ""
+  log "$1"
+  sleep 3.8
+}
+
+# Log a line to sterr and exit with error code 1
+err() {
+  echo "ERROR: $1" >&2
+  exit 1
+}
+
+# Check that the given command is installed and quit on error if that's not the case
+check_dependency() {
+  if [ -z $(command -v $1) ]; then
+    err "This script needs \"$1\" to be installed and be executable"
+  fi
+}
+
+# Get the name of the current git branch
+current_branch() {
+  echo $(git branch | sed -n -e 's/^\* \(.*\)/\1/p')
+}
+
+# Get the local name for canalplus's remote repository
+git_remote_name() {
+  git remote -v | grep "git@github.com:canalplus/rx-player.git" | grep "(push)" | cut -f1
+}
+
+# Check that the current branch is up-to-date with remote, errors if that's not
+# the case
+check_branch_synchronized_with_remote() {
+  checked_branch=$(current_branch)
+  checked_remote=$(git_remote_name)
+  if ! [ x"$(git rev-parse $checked_branch)" = x"$(git rev-parse $checked_remote/$checked_branch)" ]; then
+    err "The branch \"$checked_branch\" is not synchronized with the remote \"$checked_remote\". Please synchronize it first."
+  fi
+}
+
+check_dependency git
+check_dependency echo
+check_dependency npm
+check_dependency cut
+check_dependency grep
+check_dependency sed
+check_dependency sleep
+
```

Comment:
I like that it checks every requirements before attempting to create branch/publish because it could be annoying if the script fail midway